src/nft/hooks/usePermit2Approval.ts

```diff
@@ -7,26 +7,28 @@ import usePermit2Allowance, { AllowanceState } from 'hooks/usePermit2Allowance'
 import { useCallback, useMemo, useState } from 'react'
 import invariant from 'tiny-invariant'
 
-// TODO: This should be removed when the sdk is updated to include the new UR address
-const NFT_UNIVERSAL_ROUTER_MAINNET_ADDRESS = '0x4c60051384bd2d3c01bfc845cf5f4b44bcbe9de5'
-
 export default function usePermit2Approval(
-  amount?: CurrencyAmount<Token>,
-  maximumAmount?: CurrencyAmount<Token>,
-  shouldUseNftRouter?: boolean
+  amount: CurrencyAmount<Token> | undefined,
+  maximumAmount: CurrencyAmount<Token> | undefined,
+  nftUniversalRouterContractAddress?: string
```

Comment:
the first two params use `| undefined` and the last uses `?` I would stick with one consistent notation (? preferred)

Reply:
actually this is following zachs comment in the other pr `?` defines it as optional and `| undefined` requires you to pass in undefined and it not optional